src/js/utils/selection-utils.js

```diff
@@ -11,7 +11,23 @@ function comparePosition(selection) {
 
   const position = anchorNode.compareDocumentPosition(focusNode);
 
-  if (position & Node.DOCUMENT_POSITION_FOLLOWING) {
+  if (position & Node.DOCUMENT_POSITION_CONTAINS) {
+    return comparePosition({
+      focusNode: focusNode.childNodes[focusOffset],
+      focusOffset: 0,
+      anchorNode, anchorOffset
+    });
+  } else if (position & Node.DOCUMENT_POSITION_CONTAINED_BY) {
+    let offset = anchorOffset - 1;
+    if (offset < 0) {
+      offset = 0;
+    }
```

Comment:
`let offset = Math.max(0, anchorOffset - 1);`

What issues are being fixed by the recursive position comparisons?